### PR TITLE
Iter exercises tweak and addition

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

- Fixes a type issue on the 03-filter exercise
- Adds a new 09-image-diff exercise

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: docs/exercises/01-iteration/03-filter.exercise.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { TODO } from '../src/todo';
 import { run } from '../src/example';
 import { collecting } from '../src/collecting';
 
-function filterNullish<T>(iterable: Iterable<T>): Iterable<NonNullable<T>> {
+function filterNullish<T>(iterable: Iterable<T>): Array<NonNullable<T>> {
```
Comment (eventualbuddha) [inline docs/exercises/01-iteration/03-filter.exercise.ts:16]: Having it return an `Iterable` was intentional as this exercise is trying to nudge people away from always creating iterator sinks–and thus realizing all values–at every function boundary. Was it broken somehow for you?

File: docs/exercises/01-iteration/09-image-diff.exercise.ts
Comment (eventualbuddha) [inline docs/exercises/01-iteration/09-image-diff.exercise.ts:1]: This is great 🎉

**Review by eventualbuddha — APPROVED** [review]
Though I'd prefer to keep the filter one as-is, I'm open to changing if you think it doesn't make sense as-is.

File: docs/exercises/01-iteration/03-filter.exercise.ts (hunk 1 above)
Comment (jonahkagan) [reply docs/exercises/01-iteration/03-filter.exercise.ts:16]: Yeah, it wasn't working for me. With this solution:

Screenshot 2024-01-02 at 11 51 55 AM

I got this test failure:
Screenshot 2024-01-02 at 11 52 01 AM

In typing this comment out I ended up figuring out that the actual fix is this:

```
diff --git a/docs/exercises/01-iteration/03-filter.exercise.ts b/docs/exercises/01-iteration/03-filter.exercise.ts
index a766f7185..1263d0b32 100644
--- a/docs/exercises/01-iteration/03-filter.exercise.ts
+++ b/docs/exercises/01-iteration/03-filter.exercise.ts
@@ -34,7 +34,7 @@ function filterNullishReference<T>(
 run({
   makeInput: () => [1, 2, null, 3, undefined, 4],
   referenceImplementation: filterNullishReference,
-  exerciseImplementation: filterNullish,
+  exerciseImplementation: collecting(filterNullish),
   solutionImplementation: collecting(filterNullishSolution),
 });
```

Comment (eventualbuddha) [reply docs/exercises/01-iteration/03-filter.exercise.ts:16]: It seems to work if you make all of them use `collecting`:

```ts
run({
  makeInput: () => [1, 2, null, 3, undefined, 4],
  referenceImplementation: collecting(filterNullishReference),
  exerciseImplementation: collecting(filterNullish),
  solutionImplementation: collecting(filterNullishSolution),
});
```

Comment (jonahkagan) [reply docs/exercises/01-iteration/03-filter.exercise.ts:16]: Nice! Fixed and fixuped

**Pull request merged**